### PR TITLE
Change chat filter bypassing

### PR DESCRIPTION
**Pull request by Teremiare**

### CODE REVIEW

**Comment by Zarel** [comment]
I kind of prefer it this way. Authority being subject to their own rules is important for making sure authority doesn't turn bad?

**Comment by sparkychildcharlie** [comment]
This would be the same as staff bypassing the lobby bold, greentext filters as well as bypassing CAPs and stretchingggg filters https://github.com/Zarel/Pokemon-Showdown/blob/master/chat.js#L321

A reason why staff should be able to say banwords is to explain to the ~~trolls~~ chat why a certain phrase is banned.

However, if Zarle wants staff to not be able to bypass, maybe this should be changed for the other 2 moderation aspects as well.

**Comment by Teremiare** [comment]
Currently auth can bypass banwords when they're said, but not when having them in the username. If there happens to be a banword that an auth has in their username for whatever reason it's effectively a permanent mute. And as mentioned above it should preferably be the same for all cases.

**Comment by Zarel** [comment]
Hm, maybe this should be dealt with on a case-by-case basis

- banwords in username: definitely no, you do not want to block people from highlighting you, permanent mute is good
- banwords in text: I can understand explaining why it's banned, I guess
- greentext: no, staff should not be able to bypass
- caps filter: no, use /announce for attention
- stretch filter: no, I'm also not seeing a good use for this

**Comment by Teremiare** [comment]
Changed it to so staff no longer bypass stretching and caps, they do however still bypass slowchat (I don't really have on opinion on that one). I assume the bold and greentext ones are lobby only.

**Comment by panpawn** [comment]
We should use ``bypassall`` for stretching/caps imo

**Comment by Zarel** [comment]
^

**Comment by Teremiare** [comment]
Added bypassall checks

File: chat.js [hunk 1]
```diff
@@ -309,20 +309,22 @@ class CommandContext {
...
 
-		if (room.filterStretching && user.name.match(/(.+?)\1{5,}/i)) {
-			return this.errorReply(`Your username contains too much stretching, which this room doesn't allow.`);
-		}
-		if (room.filterCaps && user.name.match(/[A-Z\s]{6,}/)) {
-			return this.errorReply(`Your username contains too many capital letters, which this room doesn't allow.`);
-		}
-		// Removes extra spaces and null characters
-		message = message.trim().replace(/[ \u0000\u200B-\u200F]+/g, ' ');
+		if (!user.can('bypassall')) {
```
Comment (Zarel) [inline chat.js]: if (user.can('bypassall')) return true;

if you're going to do it that way

**Comment by Teremiare** [comment]
Done

**Pull request merged**